core/controllers/dashboard_test.py

```diff
@@ -66,6 +75,349 @@ def test_logged_in_notifications_dashboard(self):
+
+    def _start_computation_job(self):
+        (user_jobs_continuous_test.ModifiedUserStatsAggregator.
+         start_computation())
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                queue_name=taskqueue_services.QUEUE_NAME_DEFAULT),
+            1)
+        self.process_and_flush_pending_tasks()
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                queue_name=taskqueue_services.QUEUE_NAME_DEFAULT),
+            0)
+        self.process_and_flush_pending_tasks()
+
+    def _start_job_to_compute_num_plays(self):
+        (stats_jobs_continuous_test.ModifiedStatisticsAggregator
+         .start_computation())
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                queue_name=taskqueue_services.QUEUE_NAME_DEFAULT),
+            1)
+        self.process_and_flush_pending_tasks()
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                queue_name=taskqueue_services.QUEUE_NAME_DEFAULT),
+            0)
+        self.process_and_flush_pending_tasks()
+
+    def _get_default_value_of_impact_score(self):
+        return 0.0
+
+    def _get_default_version_of_exploration(self):
+        return 1
+
+    def test_stats_no_explorations(self):
+        self.login(self.OWNER_EMAIL_1)
+        response = self.get_json(feconf.DASHBOARD_DATA_URL)
+        self.assertEqual(response['explorations_list'], [])
+        self._start_computation_job()
+        self.assertIsNone(user_models.UserStatsModel.get(
+            self.owner_id_1, strict=False))
+        self.logout()
+
+    def test_one_play_for_single_exploration(self):
+        exploration = self.save_new_default_exploration(
+            self.EXP_ID_1, self.owner_id_1, title=self.EXP_TITLE_1)
+
+        self.login(self.OWNER_EMAIL_1)
+        response = self.get_json(feconf.DASHBOARD_DATA_URL)
+        self.assertEqual(len(response['explorations_list']), 1)
+
+        exp_version = self._get_default_version_of_exploration()
+        exp_id = self.EXP_ID_1
+        state = exploration.init_state_name
+
+        self._record_start(exp_id, exp_version, state)
+        self._start_job_to_compute_num_plays()
+
+        self._start_computation_job()
+        user_model = user_models.UserStatsModel.get(
+            self.owner_id_1, strict=False)
+        self.assertEquals(user_model.total_plays, 1)
+        self.assertEquals(
+            user_model.impact_score, self._get_default_value_of_impact_score())
+        self.assertIsNone(user_model.average_ratings)
+        self.logout()
+
+    def test_one_rating_for_single_exploration(self):
+        self.save_new_default_exploration(
+            self.EXP_ID_1, self.owner_id_1, title=self.EXP_TITLE_1)
+
+        self.login(self.OWNER_EMAIL_1)
+        response = self.get_json(feconf.DASHBOARD_DATA_URL)
+        self.assertEqual(len(response['explorations_list']), 1)
+
+        exp_id = self.EXP_ID_1
+        rating_start = 4
```

Comment:
Here and elsewhere: you don't need this to be a separate variable, especially since you don't use it anywhere else. Just inline it with the method call.